otter/integration/tests/test_convergence.py

```diff
@@ -183,6 +344,25 @@ def _choose_half_the_servers(self, (code, response)):
         self.n_killed = self.n_servers / 2
         return ids[:self.n_killed]
 
+    def _choose_servers_from_active_list(self, (code, response),
+                                         num_servers=None):
+        """Select the first 'num_servers' of the servers returned by the
+        ``get_scaling_group_state`` function.  Record the number of servers
+        received in ``active_servers`` attribute. If no num_servers is
+        provided, all active servers will be returned.
+
+        If the number of servers requested is less than the number active,
+        an error will be raised.
+        """
+
+        if code != 200:
+            raise Exception("Not 200!; Asking for the state didn't work.")
+        ids = extract_active_ids(response)
+        self.active_servers = len(ids)
```

Comment:
Out of curiosity, since, this seems to be the same kind of function as `_choose_random_servers` and `_choose_half_the_servers`, any reason this is recording in a different attribute name?  (Just wondering because accessing different attributes based on which choice function was used is a little hard to remember).

Also wondering if grabbing the first N are significant?  There is no contract that specifies what order the servers in the active list are in, so wondering if `_choose_random_servers` would be sufficient?